### PR TITLE
Enable Liquibase includeAll in Native Image

### LINKED ISSUE #16292: Liquibase includeAll does not work in native mode
This is a follow-up to #14751 which contains all the necessary info, especially https://github.com/quarkusio/quarkus/issues/14751#issuecomment-808919081

Short story: includeAll was fixed for JVM mode by upgrading to Liquibase to 4.3.2 but the classpath resource scanning in Liquibase needs some adjustments (in Quarkus) to make it work in native.

### PR DESCRIPTION
**Pull request by gcw-it**
Fixes #16292

This patch creates a `FileSystem` in the native image to enable access to the Liquibase changelog resources.

### CODE REVIEW

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/NativeImageResourceAccessor.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Collections;
+import java.util.List;
+
+import org.jboss.logging.Logger;
+
+import liquibase.resource.AbstractPathResourceAccessor;
+import liquibase.resource.PathResource;
+import liquibase.resource.Resource;
+
+public class NativeImageResourceAccessor extends AbstractPathResourceAccessor {
```
Comment (gsmet) [inline extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/NativeImageResourceAccessor.java:18]: This one needs to be in the `runtime` subpackage as it's not API.

**Review by gsmet — DISMISSED** [review]
Thanks for this PR! I added a small nitpick about packages but I will let @zakkak have a look as I'm not very familiar with this GraalVM feature.

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/NativeImageResourceAccessor.java (hunk 1 above)
Comment (gcw-it) [reply extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/NativeImageResourceAccessor.java]: done

**Review by zakkak — APPROVED** [review]
LGTM.

Please have a look at the suggestion regarding the method detecting if we are in a native image.

File: extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/LiquibaseFactory.java [hunk 2]
```diff
@@ -38,33 +41,46 @@ public LiquibaseFactory(LiquibaseConfig config, DataSource datasource, String da
     }
 
     private ResourceAccessor resolveResourceAccessor() throws FileNotFoundException {
+        var rootAccessor = new CompositeResourceAccessor();
+        return isNativeImage() ? nativeImageResourceAccessor(rootAccessor) : defaultResourceAccessor(rootAccessor);
+    }
+
+    private boolean isNativeImage() {
+        return System.getProperty(PROPERTY_IMAGE_CODE_KEY) != null;
+    }
```
[the remark this answers is not in the document]
Comment (gcw-it) [reply extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/LiquibaseFactory.java:50]: Thanks for the pointer. This seems to be the idiomatic way. I'll change the code accordingly.

**Review by zakkak — APPROVED** [review]
@gcw-it thank you. Could you please also squash the two commits?

**Comment by gcw-it** [comment]
Done.

**Pull request merged**